File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        best = nums[0]
        prefixBest = nums[0]
        for num in nums[1:]:
            prefixBest = max(prefixBest+num,num)
            best = max(best,prefixBest)
        return best

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int best          = nums[0];
        int bestEndingAti = nums[0];
        for (int i=1; i<(int)nums.size(); ++i){
            bestEndingAti = max(nums[i],bestEndingAti+nums[i]);
            best = max(best, bestEndingAti);
        }
        return best;
    }
};